src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.GroupedAnalyzerActions.cs

```diff
@@ -74,7 +86,25 @@ IGroupedAnalyzerActions IGroupedAnalyzerActions.Append(IGroupedAnalyzerActions i
 
                 var newGroupedActions = GroupedActionsByAnalyzer.AddRange(groupedAnalyzerActions.GroupedActionsByAnalyzer);
                 var newAnalyzerActions = AnalyzerActions.Append(groupedAnalyzerActions.AnalyzerActions);
-                return new GroupedAnalyzerActions(newGroupedActions, newAnalyzerActions);
+                var analyzersByKind = CreateAnalyzersByKind(newGroupedActions);
+                return new GroupedAnalyzerActions(newGroupedActions, analyzersByKind, newAnalyzerActions);
+            }
+
+            private static ImmutableSegmentedDictionary<TLanguageKindEnum, ImmutableArray<DiagnosticAnalyzer>> CreateAnalyzersByKind(ImmutableArray<(DiagnosticAnalyzer, GroupedAnalyzerActionsForAnalyzer)> groupedActionsAndAnalyzers)
+            {
+                var analyzersByKind = PooledDictionary<TLanguageKindEnum, ArrayBuilder<DiagnosticAnalyzer>>.GetInstance();
+                foreach (var (analyzer, groupedActionsForAnalyzer) in groupedActionsAndAnalyzers)
+                {
+                    foreach (var (kind, _) in groupedActionsForAnalyzer.NodeActionsByAnalyzerAndKind)
+                    {
+                        analyzersByKind.AddPooled(kind, analyzer);
```

Comment:
consider changing AddPooled to work on a Dict not an IDict (done in my pr as well, but nice to potentially pull into this pr).

Reply:
done!